### PR TITLE
MINIFICPP-1171 - Fix "%ll" format strings in log lines

### PR DESCRIPTION
**Pull request by arpadboda**
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: libminifi/src/RemoteProcessorGroupPort.cpp [hunk 1]
```diff
@@ -360,7 +360,7 @@ std::pair<std::string, int> RemoteProcessorGroupPort::refreshRemoteSite2SiteInfo
           return std::make_pair(host, siteTosite_port_);
         }
       } else {
-        logger_->log_error("Cannot output body to content for ProcessGroup::refreshRemoteSite2SiteInfo: received HTTP code %ll from %s", client->getResponseCode(), fullUrl.str());
+        logger_->log_error("Cannot output body to content for ProcessGroup::refreshRemoteSite2SiteInfo: received HTTP code %lld from %s", client->getResponseCode(), fullUrl.str());
```
Comment (szaszm) [inline libminifi/src/RemoteProcessorGroupPort.cpp]: Use `PRId64` from `<cinttypes>` for `int64_t`. `lld` is meant to be used for `long long int`, which is not guaranteed to be the same as `int64_t`. It is not the same on my platform.

File: libminifi/src/ThreadedSchedulingAgent.cpp [hunk 2]
```diff
@@ -47,15 +47,15 @@ void ThreadedSchedulingAgent::schedule(std::shared_ptr<core::Processor> processo
   if (configure_->get(Configure::nifi_administrative_yield_duration, yieldValue)) {
     core::TimeUnit unit;
     if (core::Property::StringToTime(yieldValue, admin_yield_duration_, unit) && core::Property::ConvertTimeUnitToMS(admin_yield_duration_, unit, admin_yield_duration_)) {
-      logger_->log_debug("nifi_administrative_yield_duration: [%ll] ms", admin_yield_duration_);
+      logger_->log_debug("nifi_administrative_yield_duration: [%lld] ms", admin_yield_duration_);
```
Comment (szaszm) [inline libminifi/src/ThreadedSchedulingAgent.cpp]: Use `PRId64` from `<cinttypes>` for `int64_t`. `lld` is meant to be used for `long long int`, which is not guaranteed to be the same as `int64_t`. It is not the same on my platform.

The above applies to both changes in this file.

File: libminifi/src/core/ProcessSession.cpp [hunk 3]
```diff
@@ -184,7 +184,7 @@ std::shared_ptr<core::FlowFile> ProcessSession::clone(const std::shared_ptr<core
     if (parent->getResourceClaim()) {
       if ((uint64_t) (offset + size) > parent->getSize()) {
         // Set offset and size
-        logger_->log_error("clone offset %ll and size %ll exceed parent size %llu", offset, size, parent->getSize());
+        logger_->log_error("clone offset %lld and size %lld exceed parent size %llu", offset, size, parent->getSize());
```
Comment (szaszm) [inline libminifi/src/core/ProcessSession.cpp]: Use `PRId64` from `<cinttypes>` for `int64_t`. `lld` is meant to be used for `long long int`, which is not guaranteed to be the same as `int64_t`. It is not the same on my platform.

Could you please fix the existing `llu` format specifier in the same fashion, to `PRIu64`? I know it's old code and our codebase is full of this kind of misuse, but if we already change the line, let's apply easy fixes to existing issues.

File: libminifi/src/core/yaml/YamlConfiguration.cpp [hunk 4]
```diff
@@ -195,20 +195,20 @@ void YamlConfiguration::parseProcessorNodeYaml(YAML::Node processorsNode, core::
 
         if (procCfg.schedulingStrategy == "TIMER_DRIVEN" || procCfg.schedulingStrategy == "EVENT_DRIVEN") {
           if (core::Property::StringToTime(procCfg.schedulingPeriod, schedulingPeriod, unit) && core::Property::ConvertTimeUnitToNS(schedulingPeriod, unit, schedulingPeriod)) {
-            logger_->log_debug("convert: parseProcessorNode: schedulingPeriod => [%ll] ns", schedulingPeriod);
+            logger_->log_debug("convert: parseProcessorNode: schedulingPeriod => [%lld] ns", schedulingPeriod);
```
Comment (szaszm) [inline libminifi/src/core/yaml/YamlConfiguration.cpp]: Use `PRId64` from `<cinttypes>` for `int64_t`. `lld` is meant to be used for `long long int`, which is not guaranteed to be the same as `int64_t`. It is not the same on my platform.

The above applies to all 7 changes in this file.

File: extensions/mqtt/processors/AbstractMQTTProcessor.cpp [hunk 5]
```diff
@@ -92,22 +92,22 @@ void AbstractMQTTProcessor::onSchedule(const std::shared_ptr<core::ProcessContex
     core::TimeUnit unit;
     if (core::Property::StringToTime(value, valInt, unit) && core::Property::ConvertTimeUnitToMS(valInt, unit, valInt)) {
       keepAliveInterval_ = valInt/1000;
-      logger_->log_debug("AbstractMQTTProcessor: KeepLiveInterval [%ll]", keepAliveInterval_);
+      logger_->log_debug("AbstractMQTTProcessor: KeepLiveInterval [%lld]", keepAliveInterval_);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply extensions/mqtt/processors/AbstractMQTTProcessor.cpp]: fixed by da22085

File: extensions/mqtt/processors/ConsumeMQTT.cpp [hunk 6]
```diff
@@ -70,12 +70,12 @@ void ConsumeMQTT::onSchedule(const std::shared_ptr<core::ProcessContext> &contex
   value = "";
   if (context->getProperty(QueueBufferMaxMessage.getName(), value) && !value.empty() && core::Property::StringToInt(value, valInt)) {
     maxQueueSize_ = valInt;
-    logger_->log_debug("ConsumeMQTT: Queue Max Message [%ll]", maxQueueSize_);
+    logger_->log_debug("ConsumeMQTT: Queue Max Message [%llu]", maxQueueSize_);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply extensions/mqtt/processors/ConsumeMQTT.cpp]: fixed by da22085

File: extensions/mqtt/processors/PublishMQTT.cpp [hunk 7]
```diff
@@ -58,7 +58,7 @@ void PublishMQTT::onSchedule(const std::shared_ptr<core::ProcessContext> &contex
   value = "";
   if (context->getProperty(MaxFlowSegSize.getName(), value) && !value.empty() && core::Property::StringToInt(value, valInt)) {
     max_seg_size_ = valInt;
-    logger_->log_debug("PublishMQTT: max flow segment size [%ll]", max_seg_size_);
+    logger_->log_debug("PublishMQTT: max flow segment size [%llu]", max_seg_size_);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply extensions/mqtt/processors/PublishMQTT.cpp]: fixed by da22085

File: extensions/windows-event-log/CollectorInitiatedSubscription.cpp [hunk 8]
```diff
@@ -590,7 +590,7 @@ bool CollectorInitiatedSubscription::subscribe(const std::shared_ptr<core::Proce
           if (!EvtRender(NULL, hEvent, EvtRenderEventXml, size, 0, &used, &propertyCount)) {
             if (ERROR_INSUFFICIENT_BUFFER == GetLastError()) {
               if (used > pCollectorInitiatedSubscription->maxBufferSize_.value()) {
-                logger->log_error("Dropping event %x because it couldn't be rendered within %ll bytes.", hEvent, pCollectorInitiatedSubscription->maxBufferSize_.value());
+                logger->log_error("Dropping event %p because it couldn't be rendered within %llu bytes.", hEvent, pCollectorInitiatedSubscription->maxBufferSize_.value());
```
Comment (szaszm) [inline extensions/windows-event-log/CollectorInitiatedSubscription.cpp]: :+1:

File: extensions/windows-event-log/ConsumeWindowsEventLog.cpp [hunk 9]
```diff
@@ -398,7 +398,7 @@ void ConsumeWindowsEventLog::processEvent(EVT_HANDLE hEvent) {
   if (!EvtRender(NULL, hEvent, EvtRenderEventXml, size, 0, &used, &propertyCount)) {
     if (ERROR_INSUFFICIENT_BUFFER == GetLastError()) {
       if (used > maxBufferSize_) {
-        logger_->log_error("Dropping event %x because it couldn't be rendered within %ll bytes.", hEvent, maxBufferSize_);
+        logger_->log_error("Dropping event %x because it couldn't be rendered within %llu bytes.", hEvent, maxBufferSize_);
```
Comment (szaszm) [inline extensions/windows-event-log/ConsumeWindowsEventLog.cpp]: Could you change `%x` to `%p` and `%llu` to `%" PRIu64 "`?

File: libminifi/include/FlowControlProtocol.h [hunk 10]
```diff
@@ -168,12 +168,12 @@ class FlowControlProtocol {
       logger_->log_info("NiFi Server Name %s", _serverName);
     }
     if (configure->get(Configure::nifi_server_port, value) && core::Property::StringToInt(value, _serverPort)) {
-      logger_->log_info("NiFi Server Port: [%ll]", _serverPort);
+      logger_->log_info("NiFi Server Port: [%hu]", _serverPort);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/include/FlowControlProtocol.h]: fixed by da22085

File: libminifi/src/FlowControlProtocol.cpp [hunk 11]
```diff
@@ -205,9 +205,9 @@ int FlowControlProtocol::sendRegisterReq() {
     return -1;
   }
   logger_->log_debug("Flow Control Protocol receive MsgType %s", FlowControlMsgTypeToStr((FlowControlMsgType) hdr.msgType));
-  logger_->log_debug("Flow Control Protocol receive Seq Num %ll", hdr.seqNumber);
+  logger_->log_debug("Flow Control Protocol receive Seq Num %zu", hdr.seqNumber);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/src/FlowControlProtocol.cpp]: fixed by da22085

File: libminifi/src/RemoteProcessorGroupPort.cpp (hunk 1 above)
Comment (szaszm) [reply libminifi/src/RemoteProcessorGroupPort.cpp]: fixed by da22085

File: libminifi/src/ThreadedSchedulingAgent.cpp (hunk 2 above)
Comment (szaszm) [reply libminifi/src/ThreadedSchedulingAgent.cpp]: fixed by da22085

File: libminifi/src/core/ProcessSession.cpp (hunk 3 above)
Comment (szaszm) [reply libminifi/src/core/ProcessSession.cpp]: fixed by da22085

File: libminifi/src/core/yaml/YamlConfiguration.cpp (hunk 4 above)
Comment (szaszm) [reply libminifi/src/core/yaml/YamlConfiguration.cpp]: fixed by da22085

File: libminifi/include/FlowControlProtocol.h [hunk 12]
```diff
@@ -168,12 +168,12 @@ class FlowControlProtocol {
...
     }
     if (configure->get(Configure::nifi_server_report_interval, value)) {
       core::TimeUnit unit;
       if (core::Property::StringToTime(value, _reportInterval, unit) && core::Property::ConvertTimeUnitToMS(_reportInterval, unit, _reportInterval)) {
-        logger_->log_info("NiFi server report interval: [%ll] ms", _reportInterval);
+        logger_->log_info("NiFi server report interval: [%lld] ms", _reportInterval);
```
[the remark this answers is not in the document]
Comment (szaszm) [reply libminifi/include/FlowControlProtocol.h]: fixed by da22085

File: extensions/windows-event-log/ConsumeWindowsEventLog.cpp [hunk 13]
```diff
@@ -623,7 +624,7 @@ int ConsumeWindowsEventLog::processQueue(const std::shared_ptr<core::ProcessSess
 
   auto before_time = std::chrono::high_resolution_clock::now();
   utils::ScopeGuard timeGuard([&](){
-    logger_->log_debug("processQueue processed %d Events in %llu ms",
+    logger_->log_debug("processQueue processed %d Events in %" PRIu64 " ms",
```
Comment (szaszm) [inline extensions/windows-event-log/ConsumeWindowsEventLog.cpp]: chrono durations are sadly not a good fit with printf-style format strings. The standard doesn't specify the return type of `std::chrono::milliseconds::count`, only that it's a signed integer type of at least 45 bits.

I recommend wrapping the return value in `int64_t{ ... }` and using `PRId64` as a format specifier. With list-initialization, we get a compiler error if a narrowing conversion would be necessary to convert the return value to `int64_t` and until then we can assume that we're safe and don't lose any data.

This applies to both occurences of printing durations.

Comment (szaszm) [reply extensions/windows-event-log/ConsumeWindowsEventLog.cpp]: There was a similar issue in #741: https://github.com/apache/nifi-minifi-cpp/pull/741#discussion_r386539946

Comment (szaszm) [reply extensions/windows-event-log/ConsumeWindowsEventLog.cpp]: fixed in 61107c8b

File: libminifi/include/FlowControlProtocol.h [hunk 14]
```diff
@@ -261,7 +262,7 @@ class FlowControlProtocol {
   // NiFi server Name
   std::string _serverName;
   // NiFi server port
-  int64_t _serverPort;
+  uint16_t _serverPort;
```
Comment (bakaid) [inline libminifi/include/FlowControlProtocol.h]: Yes!

**Pull request closed without merging**